configure

```diff
@@ -235,7 +224,7 @@ trap remove_tmpdir EXIT
 
 str_concat()
 {
-  echo ${ECHO_N} $@ ${ECHO_C}
+  printf '%s ' $@
```

Comment:
this line changes str_concat() behavior, adding trailing space

Reply:
Are you sure the previous implementation did not? Unfortunately one space before or after `%s` is required or else configure fails with errors like this:
```
============ Checking for linker support for -static ============

----- source file: /var/tmp/portage/app-crypt/rhash-1.3.6-r1/temp/rhash-configure--17484/tmp.c -----
int main(void) { return 0; }
----- end of file: /var/tmp/portage/app-crypt/rhash-1.3.6-r1/temp/rhash-configure--17484/tmp.c -----
x86_64-pc-linux-gnu-gcc -pipe-DNDEBUG-fomit-frame-pointer-ffunction-sections-fdata-sections -Wall -W -Wstrict-prototypes -Wnested-externs -Winline -Wpointer-arith -Wbad-function-cast -Wmissing-prototypes -Wmissing-declarations -march=native -mtune=native -O2 -pipe /var/tmp/portage/app-crypt/rhash-1.3.6-r1/temp/rhash-configure--17484/tmp.c -march=native -mtune=native -O2 -pipe -Wl,-O1 -Wl,--hash-style=gnu -Wl,--sort-common -Wl,--as-needed  -o /var/tmp/portage/app-crypt/rhash-1.3.6-r1/temp/rhash-configure--17484/tmp -static
x86_64-pc-linux-gnu-gcc: error: unrecognized command line option '-pipe-DNDEBUG-fomit-frame-pointer-ffunction-sections-fdata-sections'
Compilation result: 1

Result is: no 
=============================================

============ Checking for linker support for --version-script ============

----- source file: /var/tmp/portage/app-crypt/rhash-1.3.6-r1/temp/rhash-configure--17484/tmp.c -----
int main(void) { return 0; }
----- end of file: /var/tmp/portage/app-crypt/rhash-1.3.6-r1/temp/rhash-configure--17484/tmp.c -----
x86_64-pc-linux-gnu-gcc -pipe-DNDEBUG-fomit-frame-pointer-ffunction-sections-fdata-sections -Wall -W -Wstrict-prototypes -Wnested-externs -Winline -Wpointer-arith -Wbad-function-cast -Wmissing-prototypes -Wmissing-declarations -march=native -mtune=native -O2 -pipe /var/tmp/portage/app-crypt/rhash-1.3.6-r1/temp/rhash-configure--17484/tmp.c -march=native -mtune=native -O2 -pipe -Wl,-O1 -Wl,--hash-style=gnu -Wl,--sort-common -Wl,--as-needed  -o /var/tmp/portage/app-crypt/rhash-1.3.6-r1/temp/rhash-configure--17484/tmp -Wl,--version-script,/var/tmp/portage/app-crypt/rhash-1.3.6-r1/temp/rhash-configure--17484/tmp.txt -shared
x86_64-pc-linux-gnu-gcc: error: unrecognized command line option '-pipe-DNDEBUG-fomit-frame-pointer-ffunction-sections-fdata-sections'
Compilation result: 1

Result is: no 
=============================================

============ Checking for gettext ============

----- source file: /var/tmp/portage/app-crypt/rhash-1.3.6-r1/temp/rhash-configure--17484/tmp.c -----
#include <libintl.h>
int main(void) { return 0; }
----- end of file: /var/tmp/portage/app-crypt/rhash-1.3.6-r1/temp/rhash-configure--17484/tmp.c -----
x86_64-pc-linux-gnu-gcc -pipe-DNDEBUG-fomit-frame-pointer-ffunction-sections-fdata-sections -Wall -W -Wstrict-prototypes -Wnested-externs -Winline -Wpointer-arith -Wbad-function-cast -Wmissing-prototypes -Wmissing-declarations -march=native -mtune=native -O2 -pipe /var/tmp/portage/app-crypt/rhash-1.3.6-r1/temp/rhash-configure--17484/tmp.c -march=native -mtune=native -O2 -pipe -Wl,-O1 -Wl,--hash-style=gnu -Wl,--sort-common -Wl,--as-needed  -o /var/tmp/portage/app-crypt/rhash-1.3.6-r1/temp/rhash-configure--17484/tmp -c
x86_64-pc-linux-gnu-gcc: error: unrecognized command line option '-pipe-DNDEBUG-fomit-frame-pointer-ffunction-sections-fdata-sections'
Compilation result: 1

Result is: no 
=============================================

Error: gettext library not found
```

Would you rather prefer `printf ' %s' $@` turning it in a leading space?